src/core/Utils.re

```diff
@@ -1,2 +1,28 @@
 let foi = float_of_int;
-let iof = int_of_float;
\ No newline at end of file
+let iof = int_of_float;
+
+type scanfState = {
+  mutable accumulator: float,
+  mutable reducer: (~accumulator: float, ~value: float) => float,
+  mutable listener: float => unit,
+};
+
+type scanfRecord = {
+  start: (float => unit) => unit,
+  next: float => unit,
+};
+
+let scanf = (~reducer, ~init): scanfRecord => {
+  let state: scanfState = {accumulator: init, reducer, listener: (0.) => ()};
+
+  let start = listener => {
+    state.listener = listener;
+  };
+
+  let next = value => {
+    state.accumulator = reducer(~accumulator=state.accumulator, ~value);
+    state.listener(state.accumulator);
+  };
+
+  {start, next};
+};
```

Comment:
I'm not really sure if I'm going to use this at any point. Basically, this an accumulator for floating point numbers. I originally thought I might need it, then took more of the approach that `react-spring` does when tracking animation state – the `mover` object itself keeps track of its own `acceleration`, `velocity`, `position`, etc., which gets tracked in top level `rAF` state.